addons/account_edi_ubl_cii_tax_extension/models/account_edi_xml_ubl_20.py

```diff
@@ -0,0 +1,37 @@
+from odoo import models
+
+
+class AccountEdiXmlUBL20(models.AbstractModel):
+    _inherit = 'account.edi.xml.ubl_20'
+
+    def _get_tax_grouping_key(self, base_line, tax_data):
+        # OVERRIDE account_edi_ubl_cii
+        grouping_key = super()._get_tax_grouping_key(base_line, tax_data)
+        tax = tax_data['tax']
+
+        grouping_key |= {
+            'allowance_charge_reason_code': tax.ubl_cii_allowance_reason_code,
+            'allowance_charge_reason': tax.ubl_cii_charge_reason_code,
+        }
+
+        return grouping_key
+
+    def _get_invoice_line_allowance_vals_list(self, line, tax_values_list=None):
+        # OVERRIDE account_edi_ubl_cii
```

Comment:
Need to have a closer look to the computation but the full override is intended ? :thinking: